src/event/EventManager.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020-2023 Delft University of Technology and SynTest contributors
+ *
+ * This file is part of SynTest Framework - SynTest Core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Encoding } from "..";
+import { PluginInterface } from "./PluginInterface";
+import { ProgramState } from "./ProgramState";
+
+export class EventManager<T extends Encoding> {
+  private _state: ProgramState<T>;
+  private listeners: PluginInterface<T>[] = [];
+
+  constructor(state: ProgramState<T>) {
+    this._state = state;
+  }
+
+  registerListener(listener: PluginInterface<T>) {
+    this.listeners.push(listener);
+  }
+
+  removeListener(listener: PluginInterface<T>) {
```

Comment:
If the add method is named register, this one should be called `deregisterListener`.

Additionally, do we plan to allow users to disable the default plugins? Otherwise, this method does not have much use for a single process application

Reply:
I'll rename it.
I'd say we have some listeners (not neccessarily plugins) which could be added and removed. The "default plugins" can just be overridden through the configuration.
However, there could be a usecase for removing listeners for when you for example want a specific listener for each search target.